### PR TITLE
Fix crashes

### PR DESCRIPTION
**Pull request by bestform**

### CODE REVIEW

File: Parser/Parser.php [hunk 1]
```diff
@@ -92,4 +118,20 @@ public function parse(string $contents, string $fileName): array
...
+     */
+    private function logErrors(OutputInterface $output = null)
+    {
+        if ($output === null) {
+            return;
+        }
+
+        $output->writeln("");
+        foreach ($this->errors as $error) {
+            $output->writeln($error);
```
Comment (j6s) [inline Parser/Parser.php]: There is a definition for a `<warn>` style - this may be a good place to use it.

File: Parser/Parser.php [hunk 2]
```diff
@@ -65,18 +83,26 @@ public function parseDirectory(string $rootDirectoy, OutputInterface $output = n
...
+     * @param string $fileName
      * @return CallToContainer[]
      */
     public function parse(string $contents, string $fileName): array
     {
-        $ast = $this->parser->parse($contents);
+        try {
+            $ast = $this->parser->parse($contents);
+        } catch (Error $e) {
+            $this->errors[] = sprintf("Could not parse file %s: %s", $fileName, $e->getMessage());
```
Comment (j6s) [inline Parser/Parser.php]: Single quotes for strings (also, in a couple of other places)

File: Parser/Parser.php [hunk 3]
```diff
@@ -35,15 +44,24 @@ public function __construct()
      *
      * @param string $rootDirectoy
      * @param OutputInterface|null $output
-     * @return array
+     * @param string[] $filterDirectories
+     * @return CallToContainer[]
      */
-    public function parseDirectory(string $rootDirectoy, OutputInterface $output = null): array
+    public function parseDirectory(string $rootDirectoy, OutputInterface $output = null, $filterDirectories = ['chameleon-system/upgrade-helper']): array
```
Comment (j6s) [inline Parser/Parser.php:50]: A short explanatin of what `filterDirectories` does would help the next person reading it (e.g. it matches the full path, not a regex, matches are excluded from the result)

**Review by j6s — CHANGES_REQUESTED** [review]

File: Parser/Parser.php (hunk 1 above)
Comment (bestform) [reply Parser/Parser.php]: done

File: Parser/Parser.php (hunk 2 above)
Comment (bestform) [reply Parser/Parser.php]: done

File: Parser/Parser.php (hunk 3 above)
Comment (bestform) [reply Parser/Parser.php:50]: done

Comment (j6s) [reply Parser/Parser.php:50]: :ok:

File: Parser/Parser.php (hunk 1 above)
Comment (j6s) [reply Parser/Parser.php]: :ok:

File: Parser/Parser.php (hunk 2 above)
Comment (j6s) [reply Parser/Parser.php]: :ok:

File: Parser/Parser.php (hunk 3 above)
Comment (j6s) [reply Parser/Parser.php:50]: :ok:

**Review by j6s — APPROVED** [review]

**Pull request merged**